tutorials/commands/flux-mini-alloc.rst

```diff
@@ -0,0 +1,251 @@
+
+We ony see one job, which is our subinstance.  Where is our other subinstance?  Where are our two sleep jobs?
+
+.. note::
+
+   Depending on your terminal settings, the subinstance may be colored blue in ``flux jobs``, indicating it is a subinstance.
+
+We can see them by specifying the ``--recursive`` option to ``flux-jobs.
+
+.. code-block:: console
+
+    corona-login-node:~$ flux jobs --recursive
+           JOBID USER     NAME       ST NTASKS NNODES     TIME INFO
+     ƒgpD9HY9BsM achu     flux        R      4      4    4.74m corona[171,173-175]
+
+    ƒgpD9HY9BsM:
+        ƒge9VDjD achu     flux        R      2      2   3.199m corona[173-174]
+        ƒLU5z2zj achu     Level1      R      1      1   3.961m corona175
+
+    ƒgpD9HY9BsM/ƒge9VDjD:
+        ƒb63wFg3 achu     Level2      R      1      1   1.903m corona174
+
+From this output we can see that our first subinstance has two jobs, our
+"Level1" sleep job and a subinstance in it.  Our second subinstance has one job
+in it, our "Level2" sleep job.
+
+Another way to view this hierarchy of subinstances is via the ``flux-pstree`` command.
+
+.. code-block:: console
+
+    corona-login-node:~$ flux pstree
+    .
+    └── flux
+        ├── flux
+        │   └── Level2
+        └── Level1
+
+-------------------------------
+Subinstance via Flux Mini Batch
+-------------------------------
+
+``flux mini alloc`` is great for when you need to drop into a shell to interact
+with your subinstance.  This is often used by users to drop into a "main"
+subinstance and to interact with that subinstance.
+
+However, the majority of the time you wouldn't want this.  Most often, you want
+to launch a subinstance, perhaps launch a number of jobs within those
+subinstances, and just wait for them to complete.  The most common way to do
+this is with ``flux mini batch``.
+
+``flux mini batch`` takes a script instead of a command, so lets write two
+scripts that will do the exact same thing as we did above with ``flux mini
+alloc``.
+
+.. code-block:: sh
+
+    #!/bin/sh
+    # filename: subinstance_level1.sh
+    id1=`flux mini submit --job-name=Level1 -N1 sleep 60`
+    id2=`flux mini batch -N2 ./subinstance_level2.sh`
+    flux job status ${id1} ${id2}
+
+In this first script we are doing exactly what we did in the first
+example when we were our level 1 instance.  We first launch a sleep
+job with the name ``Level1``, with only the minor difference that I
+set the sleep time to 60 seconds instead of infinity.  We then launch
+a two node subinstance via ``flux mini batch`` and the ``-N`` option.
+This ``flux mini batch`` takes a second script as input.  We then call
+``flux job status`` to wait for the job and subinstance to finish
+before exiting the script.
+
+.. code-block:: sh
+
+    #!/bin/sh
+    # filename: subinstance_level2.sh
+    id=`flux mini submit --job-name=Level2 -N1 sleep 60`
+    flux job status ${id}
+
+In the second script (which we ran via ``flux mini batch`` in the first script),
```

Comment:
High level comment - this seems really hard, user interface-wise. If I were reading this tutorial I really wouldn't be excited to have to do this, if I had a use case. Why should I need to write nested scripts for flux, including flux commands, when I might be able to write one file with a clean logic to do it?